internal/graph/check.go

```diff
@@ -572,62 +572,177 @@ func checkAssociatedObjects(ctx context.Context, req *ResolveCheckRequest, objec
 	}, nil
 }
 
-// checkUsersetSlowPath will check userset or public wildcard path.
-// This is the slow path as it requires dispatch on all its children.
-func (c *LocalChecker) checkUsersetSlowPath(ctx context.Context, req *ResolveCheckRequest, iter *storage.ConditionsFilteredTupleKeyIterator) (*ResolveCheckResponse, error) {
-	ctx, span := tracer.Start(ctx, "checkUsersetSlowPath")
-	defer span.End()
-	var handlers []CheckHandlerFunc
-
-	typesys, _ := typesystem.TypesystemFromContext(ctx)
+type dispatchMsg struct {
+	err          error
+	shortCircuit bool
+	dispatch     CheckHandlerFunc
```

Comment:
```suggestion
	handler     CheckHandlerFunc
```

So that it's more clear that doing `dispatchMsg{dispatch: c.dispatch(ctx, req, tupleKey)` is _not_ actually calling `dispatch()`, it's merely creating a function which will be executed at a later time

We could also rename `processDispatches` to `runHandlers` and `consumeDispatches` to `waitForHandlers`

Reply:
not relevant anymore as we have dispatchParams instead as per comment below.